packages/playwright-test/src/dispatcher.ts

```diff
@@ -606,3 +607,19 @@ function chunkFromParams(params: TestOutputPayload): string | Buffer {
     return params.text;
   return Buffer.from(params.buffer!, 'base64');
 }
+
+function testResultFromSerializedInfo(serializedTestResult: SerializedTestResult, steps: TestStep[]): TestResult {
+  return {
```

Comment:
TestResult identity is important, you should always pass the same instance to the user.